js/widgets/musickeyboard.js

```diff
@@ -1,73 +1,270 @@
-// Copyright (c) 2018 Ritwik Abhishek
-// Copyright (c) 2018,20 Walter Bender
-//
-// This program is free software; you can redistribute it and/or
-// modify it under the terms of the The GNU Affero General Public
-// License as published by the Free Software Foundation; either
-// version 3 of the License, or (at your option) any later version.
-//
-// You should have received a copy of the GNU Affero General Public
-// License along with this library; if not, write to the Free Software
-// Foundation, 51 Franklin Street, Suite 500 Boston, MA 02110-1335 USA
-
-function MusicKeyboard() {
-    const FAKEBLOCKNUMBER = 100000;
-    const BUTTONDIVWIDTH = 535; // 5 buttons
-    const OUTERWINDOWWIDTH = 758;
-    const INNERWINDOWWIDTH = 50;
-    const BUTTONSIZE = 53;
-    const ICONSIZE = 32;
+/**
+ * @file This contains the prototype of the MusicKeyboard Widget.
+ *
+ * @copyright 2015 Jefferson Lee
+ * @copyright 2018 Ritwik Abhishek
+ * @copyright 2018,20 Walter Bender
+ *
+ * @license
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * The GNU Affero General Public License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with this
+ * library; if not, write to the Free Software Foundation, 51 Franklin Street, Suite 500 Boston,
+ * MA 02110-1335 USA.
+ */
+
+/*
+   global platformColor, docById, wheelnav, slicePath, logo, Singer, last, _, FIXEDSOLFEGE1
+   SHARP, FLAT, noteToFrequency, EIGHTHNOTEWIDTH, MATRIXSOLFEHEIGHT, MATRIXSOLFEWIDTH,
+   i18nSolfege, toFraction, blocks, getNote, PREVIEWVOLUME, DEFAULTVOICE, PITCHES3,
+   SOLFEGENAMES, PITCHES2, PITCHES, NOTESSHARP, NOTESFLAT, convertFromSolfege, SOLFEGECONVERSIONTABLE
+
+
+ */
+
+/* exported MusicKeyboard */
+
+/**
+ * @class
+ * @classdesc pertains to setting up all features of the musickeyboard widget and its UI features.
+ *
+ * Private members' names begin with underscore '_".
+ */
+class MusicKeyboard {
+    static FAKEBLOCKNUMBER = 100000;
+    static BUTTONDIVWIDTH = 535; // 5 buttons
+    static OUTERWINDOWWIDTH = 758;
+    static INNERWINDOWWIDTH = 50;
+    static BUTTONSIZE = 53;
+    static ICONSIZE = 32;
     // Mapping between keycodes and virtual keyboard
-    const BLACKKEYS = [87, 69, 82, 84, 89, 85, 73, 79];
-    const WHITEKEYS = [65, 83, 68, 70, 71, 72, 74, 75, 76];
-    const SPACE = 32;
+    static BLACKKEYS = [87, 69, 82, 84, 89, 85, 73, 79];
+    static WHITEKEYS = [65, 83, 68, 70, 71, 72, 74, 75, 76];
+    static SPACE = 32;
 
-    const saveOnKeyDown = document.onkeydown;
-    const saveOnKeyUp = document.onkeyup;
+    static saveOnKeyDown = document.onkeydown;
+    static saveOnKeyUp = document.onkeyup;
+    static beginnerMode = localStorage.beginnerMode;
+    static unit = MusicKeyboard.beginnerMode === "true" ? 8 : 16;
+    static selectedNotes = [];
```

Comment:
I don't think variables that do not have any meaning without instantiation of the widget (for example - `selectedNotes`) should be made static. Please consider making these instance variables instead in the constructor. Thanks